heartbeat/powervs-subnet.in

```diff
@@ -0,0 +1,1059 @@
+#!@PYTHON@ -tt
+# ------------------------------------------------------------------------
+# Description:  Resource Agent to move a Power Virtual Server subnet
+#               and its IP address from one virtual server instance
+#               to another.
+#
+# Authors:      Edmund Haefele
+#               Walter Orb
+#
+# Copyright (c) 2024 International Business Machines, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------
+
+import ipaddress
+import json
+import os
+import re
+import socket
+import subprocess
+import sys
+import textwrap
+import time
+
+import requests
+import requests.adapters
+import urllib3.util
+
+OCF_FUNCTIONS_DIR = os.environ.get(
+    "OCF_FUNCTIONS_DIR", "%s/lib/heartbeat" % os.environ.get("OCF_ROOT")
+)
+
+sys.path.append(OCF_FUNCTIONS_DIR)
+
+try:
+    import ocf
+except ImportError:
+    sys.stderr.write("ImportError: ocf module import failed.")
+    sys.exit(5)
+
+
+class PowerCloudAPIError(Exception):
+    def __init__(self, message, exit_code):
+        ocf.ocf_exit_reason(message)
+        sys.exit(exit_code)
+
+
+class nmcli:
+    """A wrapper class to run nmcli system commands."""
+
+    NMCLI_SYSTEM_CMD = ["nmcli", "-t"]
+    CONN_PREFIX = "VIP_"
+    DEV_PREFIX = "env"
+    ROUTING_PRIO = 50
+    ROUTING_TABLE = 500
+    _WAIT_FOR_NIC_SLEEP = 3
+
+    def __init__(self):
+        """Class implements only classmethods or staticmethods, instantiation is not used."""
+        pass
+
+    @classmethod
+    def _nmcli_os_cmd(cls, nmcli_args):
+        """run os nmcli command with the specified arguments.
+
+        Returns the output as a dictionary.
+        """
+
+        ocf.logger.debug(f"_nmcli_os_cmd: args: {nmcli_args}")
```

Comment:
Debian testing is failing on this line for some reason:
```
11:36:24  OCF_ROOT=. OCF_FUNCTIONS_DIR=/srv/workspace/resource-agents-pipeline_PR-1948/resource-agents/heartbeat /srv/workspace/resource-agents-pipeline_PR-1948/resource-agents/heartbeat/powervs-subnet meta-data > metadata-powervs-subnet.xml
11:36:24    File "/srv/workspace/resource-agents-pipeline_PR-1948/resource-agents/heartbeat/powervs-subnet", line 79
11:36:24      ocf.logger.debug(f"_nmcli_os_cmd: args: {nmcli_args}")
11:36:24                                                          ^
11:36:24  SyntaxError: invalid syntax
```

Reply:
nmcli_args is a list [], 
could this cause trouble in the f-string on older Python 3 (?)

Reply:
That might be why. Maybe `"{}".format(nmcli_args)` will work. Try that first, and if that doesnt work we'll have to use `.format(str(nmcli_args))`.

Reply:
We're getting closer.

It is now it failing on {e.stderr}:
```
13:06:01    File "/srv/workspace/resource-agents-pipeline_PR-1948/resource-agents/heartbeat/powervs-subnet", line 98
13:06:01      f"_nmcli_os_cmd: error executing nmcli: {e.stderr}",
13:06:01                                                        ^
13:06:01  SyntaxError: invalid syntax
```

Reply:
I found the issue.

You can probably undo the changes you've done, and update this line in configure.ac from 2.7 to 3.6:
AS_VERSION_COMPARE([$PYTHON_VERSION], [2.7], [BUILD_OCF_PY=0], [BUILD_OCF_PY=1], [BUILD_OCF_PY=1])

Reply:
I'm trying to figure what's the ideal way to implement this, so I'll send you an update as soon as I've found it.

Reply:
Let's add this before the `AC_PATH_PROG([PYTHON], [$PYTHON])` section as well:
https://github.com/ClusterLabs/fence-agents/blob/main/configure.ac#L281-L285